### ISSUE TITLE
`write(::IO, ::AnnotatedChar)` does not return an Int 

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/julia#54195]
`write` is supposed to return the number of bytes written.

```julia
julia> write(IOContext(Base.AnnotatedIOBuffer()), Base.AnnotatedChar('a', [:y => 2]))

julia> write(IOContext(Base.AnnotatedIOBuffer()), Base.AnnotatedChar('a', [:y => 2])) === nothing
true
```

#53715 will fix this particular case, but not fix the underlying issue with the `write(::IO, ::AnnotatedChar)` method.

**Comment by LilithHafner** [comment]
This is fixed on master, just needs a backport.

**Comment by KristofferC** [comment]
Which one?

**Comment by LilithHafner** [comment]
Probably a StyledStrings bump; idk. cc @tecosaur

**Comment by tecosaur** [comment]
Yep, this was fixed in https://github.com/JuliaLang/StyledStrings.jl/commit/c0847180dbb889db4ae0be2ad5becfc9d5f85787, and will be fixed in 1.11 once #54127 is merged.

**Comment by KristofferC** [comment]
(For posterity, I would suggest that you just merge those PRs yourself because if it doesn't have a backport label it is easy it get missed).
